src/wolfgame.py

```diff
@@ -6936,19 +6934,43 @@ def fspectate(var, wrapper, message):
     else:
         players = []
         if what == "wolfchat":
+            already_spectating = wrapper.source.nick in var.SPECTATING_WOLFCHAT
             var.SPECTATING_WOLFCHAT.add(wrapper.source.nick)
-            players = [p for p in list_players() if in_wolflist(p, p)]
+            players = [p for p in get_players() if in_wolflist(p.nick, p.nick)]
+            if spectate_warning and not already_spectating and var.SPECTATE_NOTICE:
+                spectator = wrapper.source.nick if var.SPECTATE_NOTICE_USER else "Someone"
+                for player in players:
+                    player.queue_message(messages["fspectate_notice"].format(spectator, what))
+                if players:
+                    player.send_messages()
         elif var.ENABLE_DEADCHAT:
+            already_spectating = wrapper.source in var.SPECTATING_DEADCHAT
             if wrapper.source in var.DEADCHAT_PLAYERS:
                 wrapper.pm(messages["fspectate_in_deadchat"])
                 return
             var.SPECTATING_DEADCHAT.add(wrapper.source)
-            players = [user.nick for user in var.DEADCHAT_PLAYERS]
+            players = var.DEADCHAT_PLAYERS
+            if spectate_warning and not already_spectating and var.SPECTATE_NOTICE:
+                spectator = wrapper.source.nick if var.SPECTATE_NOTICE_USER else "Someone"
+                for player in players:
+                    player.queue_message(messages["fspectate_notice"].format(spectator, what))
+                if players:
+                    player.send_messages()
         else:
             wrapper.pm(messages["fspectate_deadchat_disabled"])
             return
         wrapper.pm(messages["fspectate_on"].format(what))
-        wrapper.pm("People in {0}: {1}".format(what, ", ".join(players)))
+        wrapper.pm("People in {0}: {1}".format(what, ", ".join([player.nick for player in players])))
+
+@command("spectate", flag="p", pm=True, phases=("day", "night"))
+def fspectate(var, wrapper, message):
+    """Spectate wolfchat or deadchat."""
+    spectate_chat(var, wrapper, message, True)
```

Comment:
specify the True (and False in fspectate below) as a kwarg so that it is clear what that bool actually means

Reply:
Also notifying people when spectating deadchat is dumb. Better to just not let non-admins spectate deadchat (other users can simply join it)